src/main/scala/org/broadinstitute/dsde/workbench/sam/service/ResourceService.scala

```diff
@@ -324,9 +325,7 @@ class ResourceService(private val resourceTypes: Map[ResourceTypeName, ResourceT
   }
 
   private def makeValidatablePolicies(policies: Map[AccessPolicyName, AccessPolicyMembership]): Future[Set[ValidatableAccessPolicy]] = {
-    Future.traverse(policies) {
-      case (accessPolicyName, accessPolicyMembership) => makeCreatablePolicy(accessPolicyName, accessPolicyMembership)
-    }.map(_.toSet)
+    policies.toList.traverse{case (accessPolicyName, accessPolicyMembership) => makeCreatablePolicy(accessPolicyName, accessPolicyMembership)}.map(_.toSet)
```

Comment:
I checked out this branch and while it compiles, intellij complains about this line. How do I make it ok with it? Also, how can I tell what traverse does? The scaladoc for Future.traverse is clear but this traverse does not have any. It feels to me that this change only serves to make the code more opaque. I.e. unless I already know what traverse does it is much harder for me to figure out what this code does.

Reply:
I'm not sure if there's a way to make IJ recognize `traverse`. But https://github.com/typelevel/cats/blob/master/core/src/main/scala/cats/Traverse.scala#L36 is the `traverse` being used.

It is not easy to find out what code is being used if you've never come across this pattern, but it's quite common in cats. I can explain better offline.

The reason for this change tho is because `Future.traverse` won't work for `Map` when `partial unification` is on

Reply:
this could be just `Future.traverse(policies.toList)` too

I understand what cats traverse does but I want support from my tools too. I would like IJ not to give me red squiggles when the compiler say it is ok and I would like scaladoc to work.

Reply:
I can change this to `Future.traverse(policies.toList)` if you prefer. But I don't think we should limit ourselves within what IJ can support.

It's known issue with IJ that it's not compatible with lots of functions cats provides because IJ uses its own presentation compiler(most of them won't be that bad if you're okay with having red squiggles in a few places).